packages/common/src/models/gradtrak.ts

```diff
@@ -0,0 +1,202 @@
+import mongoose, { Document, InferSchemaType, Schema } from "mongoose";
+
+export const GradtrakCustomEventSchema = new Schema({
+  start_time: {
+    type: String,
+    required: true,
+    trim: true,
+    alias: "start",
+  },
+  end_time: {
+    type: String,
+    required: true,
+    trim: true,
+    alias: "end",
+  },
+  title: {
+    type: String,
+    required: false,
+    alias: "name",
+  },
+  location: {
+    type: String,
+    required: false,
+    trim: true,
+    alias: "place",
+  },
+  description: {
+    type: String,
+    required: false,
+  },
+  days_of_week: {
+    type: String,
+    required: false,
+    trim: true,
+    alias: "days",
+  },
+  uni_reqs: {
+    type: [String],
+    required: false,
+    trim: true,
+  },
+  college_reqs: {
+    type: [String],
+    required: false,
+    trim: true,
+  }
+});
+
+export const selectedCourseSchema = new Schema({
+  class_ID: {
+    type: String,
+    trim: true,
+    required: true,
+  },
+  primary_section_ID: {
+    type: String,
+    trim: true,
+    required: false,
+  },
+  secondary_section_IDs: {
+    type: [String],
+    trim: true,
+    required: false,
+  },
+  uni_reqs: {
+    type: [String],
+    required: false,
+    trim: true,
+  },
+  college_reqs: {
+    type: [String],
+    required: false,
+    trim: true,
+  }
+});
+
+export const planTermSchema = new Schema(
+  {
+    name: {
+      type: String,
+      required: false,
+    },
+    courses: {
+      type: [selectedCourseSchema],
+      required: true,
+    },
+    user_email: {
+      type: String,
+      required: true,
+    },
+    term: {
+      type: {
+        year: {
+          type: Number,
+          required: true,
+        },
+        planTerm: {
+          type: String,
+          required: true,
+          trim: true,
+        },
+      },
+      required: false,
+    },
+    custom_events: {
+      type: [GradtrakCustomEventSchema],
+      required: false,
+    },
+  }
+);
+
+export const majorReqSchema = new Schema({
+  name: {
+    type: String,
+    trim: true,
+    required: true,
+  },
+  major: {
+    type: String,
+    trim: true,
+    required: true,
+  },
+  num_courses_required: {
+    type: Number,
+    required: true,
+  },
+  satisfying_course_ids: {
+    type: [String],
+    trim: true,
+    required: false,
+  },
+  is_minor: {
+    type: Boolean,
+    required: false
+  }
+});
+
+export const gradtrakSchema = new Schema(
+  {
+    user_email: {
```

Comment:
I would stick with a standard for casing, and I'd suggest `camelCase` instead of `snake_case`.